sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -110,19 +110,25 @@ case class CreateViewCommand(
 
   private def isTemporary = viewType == LocalTempView || viewType == GlobalTempView
 
-  // Disallows 'CREATE TEMPORARY VIEW IF NOT EXISTS' to be consistent with 'CREATE TEMPORARY TABLE'
-  if (allowExisting && isTemporary) {
-    throw new AnalysisException(
-      "It is not allowed to define a TEMPORARY view with IF NOT EXISTS.")
-  }
+  if (isTemporary) verifyTempView()
+
+  private def verifyTempView(): Unit = {
+    // Disallows 'CREATE TEMPORARY VIEW IF NOT EXISTS'
+    // to be consistent with 'CREATE TEMPORARY TABLE'
+    if (allowExisting) {
+      throw new AnalysisException(
+        "It is not allowed to define a TEMPORARY view with IF NOT EXISTS.")
+    }
 
-  // Temporary view names should NOT contain database prefix like "database.table"
-  if (isTemporary && name.database.isDefined) {
-    val database = name.database.get
-    throw new AnalysisException(
-      s"It is not allowed to add database prefix `$database` for the TEMPORARY view name.")
+    // Temporary view names should NOT contain database prefix like "database.table"
+    name.database.foreach{
+      database => throw new AnalysisException(
+        s"It is not allowed to add database prefix `$database` for the TEMPORARY view name.")
+    }
   }
 
+  private var isTempReferred = false
```

Comment:
I'm not sure about the semantics of this. It's used in a different place from where it's checked. Is this always set to true by a code path before it needs to be used?

Reply:
`isTempReferred` A flag which will be true when permanent view is based on temporary view while using pgSQL dialect.

Reply:
Yes that's not quite my question. How do we know the code path that sets this to true below will execute first?

Reply:
In `run()`, before using `isTempReferred`, a call to `verifyTemporaryObjectsNotExists()` is made which will update it's value.